File: BIZ/OA/workTime/logic/code/dsrc/KSQLProcedure.java


import java.util.Map;

import com.justep.model.ModelUtils;
import com.justep.system.data.KSQL;
import com.justep.system.data.Table;
import com.justep.util.Utils;

public class KSQLProcedure {

	public static Table query(String ksql, Map<String, Object> variables, String dataModel, String fnModel) {
		Utils.check(Utils.isNotEmptyString(ksql) && Utils.isNotEmptyString(dataModel), "ksqlQueryAction中的ksql和dataModel参数必须指定！");
		return KSQL.select(ksql, variables, dataModel, Utils.isEmptyString(fnModel) ? null : ModelUtils.getModel(fnModel));

	}
	
	public static Integer update(String ksql, Map<String, Object> variables, String dataModel, String fnModel) {
		Utils.check(Utils.isNotEmptyString(ksql) && Utils.isNotEmptyString(dataModel), "ksqlUpdateAction中的ksql和dataModel参数必须指定！");
		return KSQL.executeUpdate(ksql, variables, dataModel, Utils.isEmptyString(fnModel) ? null : ModelUtils.getModel(fnModel));
	}
}

File: BIZ/OA/workTime/logic/action/ksql.procedure.m
<model xmlns="http://www.justep.com/model"
	xmlns:m="http://www.justep.com/model">
	<procedure name="ksqlUpdateProcedure"
		code-model="/OA/workTime/logic/code"
		code="KSQLProcedure.update">
		<parameter name="ksql" type="String"/>
		<parameter name="variables" type="Map"/>
		<parameter name="dataModel" type="String"/>
		<parameter name="fnModel" type="String"/>
	</procedure>
</model>